File: Samples/IngeoSDK-iOS-Sample01/iOS-Sample01/ViewController.m
//
//  ViewController.m
//  iOS-Sample01
//
//  Created by Amit Palomo on 12/29/14.
//  Copyright (c) 2014 Ingeo. All rights reserved.
//

#import "ViewController.h"
#import <IngeoSDK/IngeoSDK.h>
#import <MapKit/MapKit.h>

@interface ViewController () <IGLocationManagerDelegate>

@property (strong, nonatomic) MKMapView *mapView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self deployMap];
    
#error Please use your app secret API key. get it from http://ingeo.io.
    [IGLocationManager initWithDelegate:self secretAPIKey:@"APIKEYGOESHERE"];
    
    [IGLocationManager startUpdatingLocation];
    
}

- (void)deployMap
{
    _mapView = [[MKMapView alloc] initWithFrame:self.view.frame];
    _mapView.showsUserLocation = YES;
    _mapView.userTrackingMode = MKUserTrackingModeFollow;
    self.view = _mapView;
}

#pragma mark - IGLocationManagerDelegate

- (void)igLocationManager:(IGLocationManager *)manager didUpdateLocation:(IGLocation *)igLocation
{
    
    NSLog(@"didUpdateLocation: %@", [igLocation description]);
    
}

- (void)igLocationManager:(IGLocationManager *)manager didDetectMotionState:(IGMotionState)motionState
{
    
    NSLog(@"didDetectMotionState: %@",[IGLocation stringForMotionState:motionState]);
    
}


@end

File: Samples/Swift-Sample01/Swift-Sample01/ViewController.swift
//
//  ViewController.swift
//  Swift-Sample01
//
//  Created by Amit Palomo on 15/12/2015.
//  Copyright © 2015 Amit Palomo. All rights reserved.
//

import UIKit
import MapKit

class ViewController: UIViewController, IGLocationManagerDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        mapSetup()

        // Please use your app secret API key. get it at http://ingeo.io/signup.php
        IGLocationManager.initWithDelegate(self, secretAPIKey: "APIKEYGOESHERE")
        
        IGLocationManager.startUpdatingLocation()
        
    }

    func mapSetup() {
        let mapView = MKMapView.init(frame: self.view.frame)
        mapView.showsUserLocation = true
        mapView.setUserTrackingMode(MKUserTrackingMode.Follow, animated: false)
        self.view = mapView
    }

    // MARK: IGLocationManagerDelegate
    
    func igLocationManager(manager: IGLocationManager!, didUpdateLocation igLocation: IGLocation!) {
        print("didUpdateLocation: \(igLocation.description)")
    }
    
    func igLocationManager(manager: IGLocationManager!, didDetectMotionState motionState: IGMotionState) {
        print("didDetectMotionState: \(IGLocation.stringForMotionState(motionState))")
        
    }
}

